checker/src/path.rs

```diff
@@ -98,17 +98,17 @@ impl Path {
     /// be the one without the reference, or the the actual heap block, if the path binds to a heap
     /// location.
     #[logfn_inputs(DEBUG)]
-    pub fn implicit_dereference(path: Rc<Path>, environment: &Environment) -> Rc<Path> {
+    pub fn implicit_dereference(path: Rc<Path>, environment: &Environment) -> Option<Rc<Path>> {
```

Comment:
This can probably be reference (&Rc<Path>). Also the name of this function (and its comments) now seem a bit misleading.